scripts/run-py-typecheck.sh

```diff
@@ -3,6 +3,13 @@
 # Run type checking on manager Python files
 
 SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-DEPLOY_DIR=$SCRIPT_DIR/../deploy
+FSIM_DIR=$SCRIPT_DIR/..
+
+mypy --no-incremental \
+    $FSIM_DIR/deploy/awstools/ \
+    $FSIM_DIR/deploy/buildtools/ \
+    $FSIM_DIR/deploy/runtools/ \
+    $FSIM_DIR/deploy/util/ \
+    $FSIM_DIR/deploy/firesim \
+    $FSIM_DIR/.github/scripts/ci_variables.py
```

Comment:
Maybe it would be good to move the CI type checking to a different script / CI job that happens before spawning the manager instance so that any typing issues (i.e. keys not found in the `ci_env` dict) must be addressed before CI is run?

Reply:
I didn't do this because it 1. requires typing to be implemented for all CI scripts (very invasive) and 2. wanted feedback before executing on this.

Reply:
Most of the CI code should be typed right? I remember adding typing to a lot of stuff (not everything), regardless I think this is fine for now but perhaps leave that to a future PR. Personally, I have type checked all the CI code at some point, it seems to be fine mostly barring some issues I was having with libraries.

Reply:
I would really like to see mypy run against the CI scripts. If you want to divvy up the work i can help out a bit.